projects/011-analysis-book-inventory-management-with-json/python/source/data_manipulation_modules/rate_calculation.py

```diff
@@ -0,0 +1,46 @@
+from collections import defaultdict
+from typing import Any, Dict, List
+
+
+def calculate_highly_rated_books(data: List[Dict[str, Any]]) -> List[Dict]:
+    """Identify and print the titles and authors of books with a rating of 4.5 or higher."""
+    high_rated = []
+    for book in data:
+        if book["rating"] >= 4.5:
+            high_rated.append(book)
+
+    print("High rated books:")
+    for book in high_rated:
+        print(f"\nTitle: {book['title']}\nAuthor: {book['author']}\n")
+    return high_rated
+
+
+def calculate_top_rated_books(data: List[Dict[str, Any]]) -> List[Any]:
```

Comment:
this function is quite similar to the previous, is possible to group the logic in a unique (generic) function?